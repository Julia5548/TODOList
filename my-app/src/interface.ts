export interface ITodo{
    id:number | null;
    name:string;
    completed:boolean;
}