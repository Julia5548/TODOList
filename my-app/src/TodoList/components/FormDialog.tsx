import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import { connectModal, InjectedProps  } from 'redux-modal'
import { ITodo } from '../../interface';

interface Props extends InjectedProps {
    id : number | undefined;
    handlerRemove(id : number) : void;
}

const FormDialog = ({id, handlerRemove, show, handleHide }: Props) => {

            return(
                <Dialog open = {show} aria-labelledby = "form-dialog-title">
                    <DialogTitle id = "form-dialog-title">
                        Удалить задачу?
                    </DialogTitle>
                    <DialogContent>
                        <DialogContentText>
                            Вы действительно хотите удалить текущую задачу?
                        </DialogContentText>
                    </DialogContent>
                    <DialogActions>
                        <Button  color = "primary" onClick = {handleHide}>
                            Отмена
                        </Button>
                        <Button color = "primary" onClick  = {() => handlerRemove(id!)}>
                            Удалить
                        </Button>
                    </DialogActions>
                </Dialog>
            )
}
const modal  =  connectModal({
    name : 'deleteRequest'
})(FormDialog)

export default FormDialog
