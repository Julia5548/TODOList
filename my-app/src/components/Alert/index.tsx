import React, { useCallback } from "react";
import Alert from "@material-ui/lab/Alert";
import { IconButton } from "@material-ui/core";
import { Close } from "@material-ui/icons";
import { hideErrorAction } from "../../store/actions";
import { connect } from "react-redux";

interface IError{
    error_text: string;
    onClose : () => void;
}

const mapDispatchToProps = (dispatch) => {
    return({
        onClose : () => {
            dispatch(hideErrorAction())
        }
    });
}
const Alerts : React.FC<IError>= ({error_text, onClose} : IError) => {

    const handleClose = useCallback(() => {
       onClose();
    }, [onClose]);

    return(
        <Alert severity = "error"
        action ={
            <IconButton  aria-label="close"  size="small" color="secondary" onClick={handleClose}>
                <Close fontSize="small"/>
            </IconButton>
        }>
            {error_text}
        </Alert>
    );
}

export default connect(null, mapDispatchToProps)(Alerts);