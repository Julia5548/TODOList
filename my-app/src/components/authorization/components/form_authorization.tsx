import React, { FormEventHandler } from "react";
import { Field } from "redux-form";
import { IUser } from "../../../interface";
import { renderTextField } from "../../TextField";
import { Button, makeStyles, Grid } from "@material-ui/core";
import { NavLink, RouteComponentProps, withRouter } from "react-router-dom";


interface IProps extends RouteComponentProps{
    onLoginUser(user : IUser, history) : void;
    handleSubmit(submitOrEvent) : FormEventHandler;
}

const useStyles = makeStyles((theme) => ({

    submit : {
        margin: theme.spacing(3, 0, 2),
    },
}));

const Form : React.FC<IProps> = ({
    onLoginUser,
    handleSubmit,
    history
}: IProps) => {

    const classes = useStyles()

    const submit = (values: IUser) => {
        const user : IUser = {
            username : values.username,
            password : values.password,
            id : 0,
            logged_in : true
        };
        
        onLoginUser(user, history);
    }

    return(
        <form onSubmit={handleSubmit(submit)}>
            <Field name="username" component={renderTextField} label="username" />
            <Field name="password" component={renderTextField} label="password" />
            <Button
                type="submit"
                fullWidth
                variant = "contained"
                color = "primary"
                className = {classes.submit}>
                    Войти
            </Button>
            <Grid container>
                <Grid item xs>
                    <NavLink to="/reset">
                        Забыли пароль?
                    </NavLink>
                </Grid>
                <Grid item>
                    <NavLink to="/registration" >
                        Вы еще не зарегистрированы?<br/>Регистрация
                    </NavLink>
                </Grid>
            </Grid>
        </form>
    )
}

export default withRouter(Form);