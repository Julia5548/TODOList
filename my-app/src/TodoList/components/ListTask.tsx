import React, { useState } from 'react';
import DeleteIcon from '@material-ui/icons/Delete';
import { makeStyles, Typography, Card, CardContent, IconButton, Checkbox} from '@material-ui/core';
import { ITodo } from '../../interface';
import FormDialog from '../components/FormDialog';
import { RouteComponentProps } from 'react-router';
import {  withRouter } from 'react-router-dom';
import { InjectedFormProps } from 'redux-form';



const useStyles = makeStyles((theme) => ({

    root: {
        display: 'flex',
        padding : theme.spacing(0,3,0,3),
        margin : theme.spacing(5),
      },
      details: {
        display: 'flex',
        flexDirection: 'column',
      },
      notCheck :{
        flex: '1 0 auto',
      },
      check: {
        flex: '1 0 auto',
        textDecoration : 'line-through',
        color: 'rgba(0, 0, 0, 0.12)',
        boxShadow: 'none',
      },
}))

interface TodoListProps extends RouteComponentProps<{pk : string}>{
    todoList : ITodo[];
    onToggle(task : ITodo) : void;
    onRemove(task : ITodo) : void;
}

export const ListTask : React.FC<TodoListProps> = (props: TodoListProps) => {

    const classes = useStyles();
    const  {todoList, onRemove, onToggle }  = props;
    
    const[open, setOpen] = useState(false);
    const pk : any = props.match.params.pk

    const todo : ITodo = {
        id : 0,
        user : pk,
        name : '',
        completed : false,

    }
    const [task, setTask] = useState(todo)

    if(todoList.length === 0 ){
        return (
            <Typography variant = "h5" component = "h5" align = 'center'>
                Задач нет!
            </Typography>
        )
    }

    const handleClose = () => {
        setOpen(false)
    }

    const handleRemoveTask = ( task : ITodo) => {
        onRemove(task)
        setOpen(false)
    }

    const handleOpen = (task : ITodo) => {
        setOpen(true)
        setTask(task)
    }

    return(
        <div>
            {todoList.map((todo) => {
                let classCheked = classes.notCheck
                if(todo.completed){
                    classCheked = classes.check
                }
                return(
                    <Card key = {todo.id}  className = {classes.root}>
                        <Checkbox
                            checked = {todo.completed}
                            name = 'checkBox_toggle_task'
                            onChange = {() => {onToggle(todo)}}/>
                        <CardContent className = {classCheked}>
                            <Typography variant = "h6" component = "h6">
                                {todo.name}
                            </Typography>
                        </CardContent>
                        <IconButton aria-label = "delete" color="secondary" edge="end" onClick = {() => handleOpen(todo)}>
                            <DeleteIcon fontSize = "small"/>
                        </IconButton>
                        <FormDialog todo = {task} handlerRemove = {handleRemoveTask} show = {open} handleDestroy = {handleClose} handleHide = {handleClose} />
                    </Card>
                    )
                })}
        </div>
    )
}

export default withRouter(ListTask)