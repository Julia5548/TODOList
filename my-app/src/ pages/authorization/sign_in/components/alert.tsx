import React from "react";
import Alert from "@material-ui/lab/Alert";

interface IError{
    error_text: string;
}
const Alerts : React.FC<IError>= ({error_text} : IError) => {
    return(
        <Alert severity = "error">
            {error_text}
        </Alert>
    );
}

export default Alerts;