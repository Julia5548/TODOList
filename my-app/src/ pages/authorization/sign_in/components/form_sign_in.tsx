import React from "react";
import { Field, InjectedFormProps, reduxForm } from "redux-form";
import { renderTextField } from "../../../../components/TextField";
import { Button, makeStyles, Grid } from "@material-ui/core";
import { NavLink, RouteComponentProps, withRouter } from "react-router-dom";
import { IUser } from "../../../../interface";
import Alert from "../../../../components/Alert/index";


interface IProps extends RouteComponentProps{
    onLoginUser(user : IUser, history) : void;
    is_error_auth: boolean;
}

const useStyles = makeStyles((theme) => ({

    submit : {
        margin: theme.spacing(3, 0, 2),
    },
}));

const Form : React.FC<IProps & InjectedFormProps<{}, IProps>> = ({onLoginUser, is_error_auth, history , ...props}) => {

    const classes = useStyles()

    const submit = values => {
        const user : IUser = {
            username : values.username,
            password : values.password,
            id : 0,
            is_logged_in : true,
            is_error_auth : false
        };
        onLoginUser(user, history);
    }

    return(
        <form onSubmit= {props.handleSubmit(submit)}>
            <Field name="username" component={renderTextField} label="username" />
            <Field name="password" component={renderTextField} label="password" />
            <Button
                type="submit"
                fullWidth
                variant = "contained"
                color = "primary"
                className = {classes.submit}>
                    Войти
            </Button>
            {is_error_auth && <Alert error_text = {'Неверен логин или пароль'}/>}
            <Grid container>
                <Grid item xs>
                    <NavLink to="/reset">
                        Забыли пароль?
                    </NavLink>
                </Grid>
                <Grid item>
                    <NavLink to="/registration" >
                        Вы еще не зарегистрированы?<br/>Регистрация
                    </NavLink>
                </Grid>
            </Grid>
        </form>
    )
}

const form = reduxForm<{}, IProps>({
    form : 'auth'
})(Form);

export default withRouter(form);