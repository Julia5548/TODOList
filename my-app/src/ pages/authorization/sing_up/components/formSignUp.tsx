import React from "react";
import { Field, InjectedFormProps, reduxForm } from "redux-form";
import { renderTextField } from "../../../../components/TextField";
import { Button, makeStyles, Grid } from "@material-ui/core";
import { NavLink, RouteComponentProps, withRouter } from "react-router-dom";
import { IUser } from "../../../../interface";


interface IProps extends RouteComponentProps{
    onCreateUser(newUser : IUser, history) : void;
}

const useStyles = makeStyles((theme) => ({

    submit : {
        margin: theme.spacing(3, 0, 2),
    },
}));

const Form : React.FC<IProps & InjectedFormProps<{}, IProps>> = ({onCreateUser, history , ...props}) => {

    const classes = useStyles()

    const submit = values => {
        
        const user : IUser = {
            username : values.username,
            email : values.email,
            password : values.password,
            id : 0,
            is_logged_in : false,
            is_error_auth : false
        };
        onCreateUser(user, history);
    }


    return(
        <form onSubmit={props.handleSubmit(submit)}>
            <Field name="username" component={renderTextField} label="username" />
            <Field name="email" component={renderTextField} label="email" />
            <Field name="password" component={renderTextField} label="password" />
            <Button
                type="submit"
                fullWidth
                variant = "contained"
                color = "primary"
                className = {classes.submit}>
                    Зарегистрироваться
            </Button>
            <Grid 
                container
                direction="row"
                justify="flex-end"
                alignItems="baseline">
                    <NavLink to="/" >
                        Авторизация
                    </NavLink>
            </Grid>
        </form>
    )
}

const form = reduxForm<{}, IProps>({
    form : 'sign-up'
})(Form);

export default withRouter(form);