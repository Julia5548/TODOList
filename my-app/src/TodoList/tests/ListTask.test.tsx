import React from 'react'
import {ListTask} from '../components/ListTask'
import { ITodo } from '../../interface';
import { Checkbox, IconButton } from '@material-ui/core';
import DeleteIcon from '@material-ui/icons/Delete';
import { shallow } from 'enzyme';


describe('<ListTask />', () => {
    let wrapper: any;

    const todoList: ITodo[] = [{id: 1, user:1, name:'покормить кота', completed : false}]
    const onRemove = jest.fn()
    const onToggle = jest.fn()

    const props : any ={
        match : {
            params:{
                pk : 1,
            }
        } 
    };

    beforeEach(() => {
        wrapper = shallow(<ListTask todoList = {todoList} onRemove = {onRemove} onToggle = {onToggle} {...props}/>)
    })

    it('changes as checkbox is checked/unchecked', () =>{
        const checkboxTask = wrapper.find(Checkbox)

        expect(checkboxTask.prop('name')).toBe('checkBox_toggle_task')
        expect(checkboxTask.prop('checked')).toBe(todoList[0].completed)
    })
    it('renders next icon button', () => {
        const nextIconButton = wrapper.find(IconButton)

        // console.log(toJson(nextIconButton))
        expect(nextIconButton.prop('aria-label')).toBe('delete')
        expect(nextIconButton.children().find(DeleteIcon))
    })
})