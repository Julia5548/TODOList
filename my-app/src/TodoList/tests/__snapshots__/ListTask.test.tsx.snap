// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListTask /> defines card tasks changes as checkbox is checked/unchecked 1`] = `
<WithStyles(ForwardRef(Typography))
  align="center"
  component="h5"
  variant="h5"
>
  Задач нет!
</WithStyles(ForwardRef(Typography))>
`;
